File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/models/statistics/serverLoadStatistics.ts
import { BaseStatistics, LoadEvent, LoadMethodStatistics } from "../../..";

export interface ServerLoadStatistics extends BaseStatistics {
    amountOfEvents: number;
    lastEvent: LoadEvent | null;
    loadMethodStatistics: LoadMethodStatistics | null;
}

export function getDefaultServerLoadStatistics(): ServerLoadStatistics {
    return {
        id: "",
        collectedDateUTC: new Date(),
        amountOfEvents: 0,
        lastEvent: null,
        loadMethodStatistics: null,
    }
}

File: src/ServerPulse.Backend/AnalyzerApi.Core/Models/Statistics/ServerLoadStatistics.cs
using AnalyzerApi.Core.Models.Wrappers;

namespace AnalyzerApi.Core.Models.Statistics
{
    public class ServerLoadStatistics : BaseStatistics
    {
        public int AmountOfEvents { get; set; }
        public LoadEventWrapper? LastEvent { get; set; }
        public LoadMethodStatistics? LoadMethodStatistics { get; set; }
    }
}